Formula/needle.rb

```diff
@@ -14,10 +14,21 @@ class Needle < Formula
 
   depends_on xcode: ["12.2", :build]
 
+  # Fix compilation on newer Swift versions
+  patch do
+    url "https://github.com/uber/needle/commit/8f9dac6d0787ce3c23e13fefb805b6d874fce7f9.patch?full_index=1"
+    sha256 "4f1481ce33046628bcf08a38e008741464a454c1b4fead791fe764a148cbb53f"
+  end
+  patch do
+    url "https://github.com/uber/needle/commit/30a4f935bc5bf6ffe83e030b29c35409c72158d4.patch?full_index=1"
+    sha256 "b0692c29c37b0cd2d686ed1628894feedc675ebf3951ab3cf0ac8a8360f1b745"
+  end
+
   def install
     system "make", "install", "BINARY_FOLDER_PREFIX=#{prefix}"
     bin.install "./Generator/bin/needle"
     libexec.install "./Generator/bin/lib_InternalSwiftSyntaxParser.dylib"
+    deuniversalize_machos libexec/"lib_InternalSwiftSyntaxParser.dylib"
```

Comment:
Can we not patch `make` to build the correct one here?

Reply:
Makefile copies library from Xcode.app (https://github.com/uber/needle/blob/v0.17.2/Makefile#L49) and then just modifies RPATH.

Reply:
I don't think that allows us to ship it since XCode is proprietary

Reply:
We could use the one from the swift formula though. That is guaranteed open source

Reply:
I'll give that a try.

Reply:
Looks like `swift` might be taking over build from Xcode and causing some issues.